### ISSUE TITLE
centerNet from scratch

### ISSUE THREAD

**Issue by joihn** [open-mmlab/mmdetection#6597]
I'm having a lot of trouble training centerNet from scratch on a custom dataset (which only have 1 class).
The model does not converge. Bounding Box mAP is still 0 after >30 epoch.
I have succesfully trained a FCRNN from scratch, after 1 epoch BBox mAP was already > 0.1, but no success at all with centernet :(

Is there something special in this model that prevent it from being trained from scratch ?

Here is my conf. for reference.

```
# Config:
data_root = '/home/dockeruser/Documents/dataset/1RUMG'
classes = ('rumex', )
custom_hooks = [
    dict(type='NumClassCheckHook'),
    dict(type='Wandb_conf_hook', priority='VERY_LOW')
]
dataset_type = 'CocoDataset'
RUMEX_MEAN = [37.8847, 48.0685, 31.3074]
RUMEX_STD = [21.5188, 22.8212, 16.0459]
img_norm_cfg = dict(
    mean=[37.8847, 48.0685, 31.3074],
    std=[21.5188, 22.8212, 16.0459],
    to_rgb=True)
WIDTH = 1536
HEIGHT = 1152
albu_train_transforms = [
    dict(
        type='OneOf',
        transforms=[
            dict(
                type='RandomResizedCrop',
                width=1536,
                height=1152,
                scale=(0.5, 1.0),
                ratio=(0.85, 1.15),
                p=0.6),
            dict(type='Resize', width=1536, height=1152, p=0.4)
        ],
        p=1),
    dict(
        type='RandomBrightnessContrast',
        brightness_limit=0.03,
        contrast_limit=0.03,
        p=0.5),
    dict(
        type='HueSaturationValue',
        hue_shift_limit=10,
        sat_shift_limit=10,
        val_shift_limit=10,
        p=0.5),
    dict(type='Flip', p=1)
]
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='Albu',
        transforms=[
            dict(
                type='OneOf',
                transforms=[
                    dict(
                        type='RandomResizedCrop',
                        width=1536,
                        height=1152,
                        scale=(0.5, 1.0),
                        ratio=(0.85, 1.15),
                        p=0.6),
                    dict(type='Resize', width=1536, height=1152, p=0.4)
                ],
                p=1),
            dict(
                type='RandomBrightnessContrast',
                brightness_limit=0.03,
                contrast_limit=0.03,
                p=0.5),
            dict(
                type='HueSaturationValue',
                hue_shift_limit=10,
                sat_shift_limit=10,
                val_shift_limit=10,
                p=0.5),
            dict(type='Flip', p=1)
        ],
        bbox_params=dict(
            type='BboxParams',
            format='pascal_voc',
            label_fields=['gt_labels'],
            min_visibility=0.1,
            filter_lost_elements=True),
        keymap=dict(img='image', gt_bboxes='bboxes'),
        update_pad_shape=False,
        skip_img_without_anno=False),
    dict(type='ToFloat32'),
    dict(
        type='RandomCenterCropPad',
        ratios=None,
        border=None,
        mean=[0, 0, 0],
        std=[1, 1, 1],
        to_rgb=True,
        test_mode=True,
        test_pad_mode=['logical_or', 31],
        test_pad_add_pix=1),
    dict(
        type='Normalize',
        mean=[37.8847, 48.0685, 31.3074],
        std=[21.5188, 22.8212, 16.0459],
        to_rgb=True),
    dict(type='DefaultFormatBundle'),
    dict(
        type='Collect',
        keys=['img', 'gt_bboxes', 'gt_labels'],
        meta_keys=('filename', 'ori_shape', 'img_shape', 'img_norm_cfg',
                   'pad_shape', 'scale_factor'))
]
test_pipeline = [
    dict(type='LoadImageFromFile', to_float32=True),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1536, 1152),
        flip=False,
        transforms=[
            dict(type='Resize', img_scale=(1536, 1152)),
            dict(type='ToFloat32'),
            dict(
                type='RandomCenterCropPad',
                ratios=None,
                border=None,
                mean=[0, 0, 0],
                std=[1, 1, 1],
                to_rgb=True,
                test_mode=True,
                test_pad_mode=['logical_or', 31],
                test_pad_add_pix=1),
            dict(
                type='Normalize',
                mean=[37.8847, 48.0685, 31.3074],
                std=[21.5188, 22.8212, 16.0459],
                to_rgb=True),
            dict(type='DefaultFormatBundle'),
            dict(
                type='Collect',
                meta_keys=('filename', 'ori_shape', 'img_shape', 'pad_shape',
                           'scale_factor', 'img_norm_cfg', 'border'),
                keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=8,
    workers_per_gpu=5,
    train=dict(
        classes=('rumex', ),
        type='CocoDataset',
        ann_file=
        '/home/dockeruser/Documents/code/datasetStat/output/1rumex_train.json',
        img_prefix='/home/dockeruser/Documents/dataset/1RUMG',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(
                type='Albu',
                transforms=[
                    dict(
                        type='OneOf',
                        transforms=[
                            dict(
                                type='RandomResizedCrop',
                                width=1536,
                                height=1152,
                                scale=(0.5, 1.0),
                                ratio=(0.85, 1.15),
                                p=0.6),
                            dict(
                                type='Resize', width=1536, height=1152, p=0.4)
                        ],
                        p=1),
                    dict(
                        type='RandomBrightnessContrast',
                        brightness_limit=0.03,
                        contrast_limit=0.03,
                        p=0.5),
                    dict(
                        type='HueSaturationValue',
                        hue_shift_limit=10,
                        sat_shift_limit=10,
                        val_shift_limit=10,
                        p=0.5),
                    dict(type='Flip', p=1)
                ],
                bbox_params=dict(
                    type='BboxParams',
                    format='pascal_voc',
                    label_fields=['gt_labels'],
                    min_visibility=0.1,
                    filter_lost_elements=True),
                keymap=dict(img='image', gt_bboxes='bboxes'),
                update_pad_shape=False,
                skip_img_without_anno=False),
            dict(type='ToFloat32'),
            dict(
                type='RandomCenterCropPad',
                ratios=None,
                border=None,
                mean=[0, 0, 0],
                std=[1, 1, 1],
                to_rgb=True,
                test_mode=True,
                test_pad_mode=['logical_or', 31],
                test_pad_add_pix=1),
            dict(
                type='Normalize',
                mean=[37.8847, 48.0685, 31.3074],
                std=[21.5188, 22.8212, 16.0459],
                to_rgb=True),
            dict(type='DefaultFormatBundle'),
            dict(
                type='Collect',
                keys=['img', 'gt_bboxes', 'gt_labels'],
                meta_keys=('filename', 'ori_shape', 'img_shape',
                           'img_norm_cfg', 'pad_shape', 'scale_factor'))
        ],
        filter_empty_gt=False),
    val=dict(
        classes=('rumex', ),
        type='CocoDataset',
        ann_file=
        '/home/dockeruser/Documents/code/datasetStat/output/1rumex_val.json',
        img_prefix='/home/dockeruser/Documents/dataset/1RUMG',
        filter_empty_gt=False,
        pipeline=[
            dict(type='LoadImageFromFile', to_float32=True),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1536, 1152),
                flip=False,
                transforms=[
                    dict(type='Resize', img_scale=(1536, 1152)),
                    dict(type='ToFloat32'),
                    dict(
                        type='RandomCenterCropPad',
                        ratios=None,
                        border=None,
                        mean=[0, 0, 0],
                        std=[1, 1, 1],
                        to_rgb=True,
                        test_mode=True,
                        test_pad_mode=['logical_or', 31],
                        test_pad_add_pix=1),
                    dict(
                        type='Normalize',
                        mean=[37.8847, 48.0685, 31.3074],
                        std=[21.5188, 22.8212, 16.0459],
                        to_rgb=True),
                    dict(type='DefaultFormatBundle'),
                    dict(
                        type='Collect',
                        meta_keys=('filename', 'ori_shape', 'img_shape',
                                   'pad_shape', 'scale_factor', 'img_norm_cfg',
                                   'border'),
                        keys=['img'])
                ])
        ]),
    test=dict(
        classes=('rumex', ),
        type='CocoDataset',
        ann_file=
        '/home/dockeruser/Documents/code/datasetStat/output/1rumex_test.json',
        img_prefix='/home/dockeruser/Documents/dataset/1RUMG',
        filter_empty_gt=False,
        pipeline=[
            dict(type='LoadImageFromFile', to_float32=True),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1536, 1152),
                flip=False,
                transforms=[
                    dict(type='Resize', img_scale=(1536, 1152)),
                    dict(type='ToFloat32'),
                    dict(
                        type='RandomCenterCropPad',
                        ratios=None,
                        border=None,
                        mean=[0, 0, 0],
                        std=[1, 1, 1],
                        to_rgb=True,
                        test_mode=True,
                        test_pad_mode=['logical_or', 31],
                        test_pad_add_pix=1),
                    dict(
                        type='Normalize',
                        mean=[37.8847, 48.0685, 31.3074],
                        std=[21.5188, 22.8212, 16.0459],
                        to_rgb=True),
                    dict(type='DefaultFormatBundle'),
                    dict(
                        type='Collect',
                        meta_keys=('filename', 'ori_shape', 'img_shape',
                                   'pad_shape', 'scale_factor', 'img_norm_cfg',
                                   'border'),
                        keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='bbox')
checkpoint_config = dict(interval=1)
wnb_username = 'PRIVATE'
wnb_project_name = 'PRIVATE'
model_name = 'temp'
resume_wandb = False
log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(
            type='WandbLoggerHook',
            init_kwargs=dict(
                project='PRIVATE', name='temp', entity='PRIVATE'))
    ])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1), ('val', 1)]
norm_cfg = dict(type='BN', requires_grad=True)
model = dict(
    type='CenterNet',
    backbone=dict(
        type='ResNet',
        depth=18,
        norm_eval=False,
        frozen_stages=-1,
        zero_init_residual=False,
        norm_cfg=dict(type='BN', requires_grad=True),
        init_cfg=None),
    neck=dict(
        type='CTResNetNeck',
        in_channel=512,
        num_deconv_filters=(256, 128, 64),
        num_deconv_kernels=(4, 4, 4),
        use_dcn=True),
    bbox_head=dict(
        type='CenterNetHead',
        num_classes=1,
        in_channel=64,
        feat_channel=64,
        loss_center_heatmap=dict(type='GaussianFocalLoss', loss_weight=1.0),
        loss_wh=dict(type='L1Loss', loss_weight=0.1),
        loss_offset=dict(type='L1Loss', loss_weight=1.0)),
    train_cfg=None,
    test_cfg=dict(topk=100, local_maximum_kernel=3, max_per_img=100))
optimizer = dict(
    type='SGD',
    lr=0.02,
    momentum=0.9,
    weight_decay=0.0001,
    paramwise_cfg=dict(norm_decay_mult=0))
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.1,
    step=[65, 71])
runner = dict(type='EpochBasedRunner', max_epochs=73)

```

**Comment by BIGWangYuDong** [comment]
Have you tried to train your model without changing any code?

**Comment by joihn** [comment]
Yes I had originally tried it without success either. Also tried on Coco instead of my dataset, without success. I think this model has issues when training from scratch, might be linked to a vanishing gardient problem.

**Comment by alexarsenault** [comment]
I've also experienced this. Would be great to get some community support on this

**Comment by soHardToHaveAName** [comment]
It should be the learning rate is set too high, you can refer to this [https://github.com/open-mmlab/mmdetection/issues/7857](https://github.com/open-mmlab/mmdetection/issues/7857)
